### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by rreinold**
basically a rewrite

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -2,36 +2,41 @@
...
-![alt text](https://raw.githubusercontent.com/rreinold/real-time-asset-tracking-template/master/banner.png "Logo Title Text 1")
+`Point with ID: 2b is within geofence`
+
+## Assets
 
-# Contents
 ### Portals
-CJS_Portal
+
+- `CJS_Portal` - Holistic view of your Real-Time Asset Tracking System, with a 3D Map of Tags
```
Comment (yashjain28) [inline Readme.md]: Please conform to portal case conventions PascalCase

File: Readme.md [hunk 2]
```diff
@@ -42,32 +47,186 @@ CJS_Portal
...
+- `RTAssetConfiguration` - Configuration for the System
+- `RTAssetUtil` - Reusable utility methods for System
 
 ### Mesaging Topics
->1. position - Things publish their position on this topic
->2. geofence - Geofence status is published here
+- `/location` - Tags publish their position using configurable [Message](#Message) structure
+- `/geofence` - Publishes status whether a tag is inside the geofence
 
 ### Collections
```
Comment (yashjain28) [inline Readme.md:74]: Please conform to collection case conventions snake_case

File: Readme.md [hunk 3]
```diff
@@ -42,32 +47,186 @@ CJS_Portal
...
->1. MessageHistory - History of all MQTT Messages received 
->2. Tags - Latest position of all tags
+>1. `LocationLogs` - History of all MQTT Messages received 
+>2. `Tags` - Latest location of all tags
 
 ### Triggers
```
Comment (yashjain28) [inline Readme.md]: Please conform to Trigger case conventions PascalCase

File: Readme.md [hunk 4]
```diff
@@ -42,32 +47,186 @@ CJS_Portal
...
+| Param | Type | Description |
+| --- | --- | --- |
+| unparsed | <code>string</code> | message body, needs to be JSON format |
+
 
-# Development
+## Development
 
-## How to Update Graph
+### How to Update Graph
```
Comment (yashjain28) [inline Readme.md]: Isn't the API section the last one, or it's fine to add credits in this one?

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+.cbmeta
```
Comment (yashjain28) [inline .gitignore:1]: jsdoc.md
We should not be uploading jsdoc.md file.

File: data/Tags.json [hunk 6]
```diff
@@ -9,7 +9,6 @@
             "last_updated": "2017-07-31T16:22:13.331Z",
             "lat": 40.75533650753633,
             "long": -74.0036954250063,
-            "payload": "Disabled",
```
Comment (yashjain28) [inline data/Tags.json:12]: Not sure if you want to upload all the data. That's unwanted information.

File: Readme.md [hunk 7]
```diff
@@ -42,32 +47,186 @@ CJS_Portal
...
+<dd></dd>
+<dt><a href="#RTAssetConfiguration">RTAssetConfiguration</a></dt>
+<dd></dd>
+<dt><a href="#RTAssetUtil">RTAssetUtil</a></dt>
+<dd></dd>
+</dl>
+
+<a name="RTAssetCheckGeofence"></a>
+
+## RTAssetCheckGeofence()
```
Comment (yashjain28) [inline Readme.md]: All these functions are under API so should be ### or even ####
These are very big headings if you take a look at the readme.

File: Readme.md [hunk 8]
```diff
@@ -42,32 +47,186 @@ CJS_Portal
...
->1. trigger_messageToCollectionCreate
->2. trigger_messageToCollectionUpdate
->3. trigger_messageToHTTPPost
+>1. `RTAssetCheckGeofence_location` - Runs RTAssetUpdateTagLocation upon publish to /location topic
+>2. `RTAssetLogLocation_location` - Runs RTAssetLogLocation upon publish to /location topic
+>3. `RTAssetUpdateTagLocation_location` - Runs RTAssetUpdateTagLocation upon publish to /location topic
+
+## API
+
+## Functions
```
Comment (yashjain28) [inline Readme.md]: functions are inside API shouldn't it be ###

File: package.json [hunk 9]
```diff
@@ -4,13 +4,14 @@
   "description": "Real-time asset tracking template for ClearBlade Platform",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (yashjain28) [inline package.json:7]: Really liked this idea. Good one. But let me know if you just want to do Libraries now or maybe with params, like 
npm run jsdoc -services
npm run jsdoc -libraries
npm run jsdoc -all
https://stackoverflow.com/questions/35221098/passing-arguments-to-npm-script-in-package-json

**Review by yashjain28 — CHANGES_REQUESTED** [review]

File: Readme.md (hunk 2 above)
Comment (rreinold) [reply Readme.md]: these are message topics

File: Readme.md (hunk 1 above)
Comment (rreinold) [reply Readme.md]: e0c0342

File: Readme.md (hunk 2 above)
Comment (rreinold) [reply Readme.md:74]: 6d8e719

File: Readme.md (hunk 3 above)
Comment (rreinold) [reply Readme.md]: da41052

File: data/Tags.json (hunk 6 above)
Comment (rreinold) [reply data/Tags.json:12]: This data is displayed in the Portal OOB

File: .gitignore (hunk 5 above)
Comment (rreinold) [reply .gitignore:1]: cfb0ff16c336566be8bc00a9eda3ef697f9f2ec8

File: package.json (hunk 9 above)
Comment (rreinold) [reply package.json:7]: this one needs to jsdoc all, we'll tackle the flag on the next one

**Comment by rreinold** [comment]
addressed issues

File: package.json (hunk 9 above)
Comment (yashjain28) [reply package.json:7]: Np

**Comment by yashjain28** [comment]
Just one change and it can be merged then.

File: data/Tags.json (hunk 6 above)
Comment (yashjain28) [reply data/Tags.json:12]: Np

File: Readme.md (hunk 1 above)
Comment (yashjain28) [reply Readme.md]: Rename in Readme CJS_Portal and Triggers section

**Review by yashjain28 — CHANGES_REQUESTED** [review]
A couple of small changes. 
Just a heads up. We need to make changes in the roles (and portals also if we're using it there) if we rename a service. And if rename a lib then servicesName.json needs to be changed for dependencies.

**Comment by rreinold** [comment]
Roles updated 4223aeeafc29718110ca6e6b13258fdf5f9545b8

**Comment by rreinold** [comment]
e1102a0 Finished rename of portal

**Pull request merged**